Runtime 5.0.7 release notes (#1043)
astronomer/docs: astro/runtime-release-notes.md

description: Release notes for Astro Runtime, the differentiated Apache Airflow experience and execution framework.
---

<!--version-specific-->

Astro Runtime is a Docker image built and published by Astronomer that extends the Apache Airflow project to provide a differentiated data orchestration experience. This document provides a summary of changes made to each available version of Astro Runtime. Note that some changes to Runtime might be omitted based on their availability in Astro.

To upgrade Astro Runtime, see [Upgrade Astro Runtime](upgrade-runtime.md). For general product release notes, see [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, contact [Astronomer support](https://support.astronomer.io).

## Astro Runtime 5.0.7

- Release date: August 16, 2022
- Airflow version: 2.3.3

### Backported fixes from Apache Airflow

> Astro Runtime 5.0.7 includes several bug fixes that were backported from future Apache Airflow releases. Fixes include:

Reviewer: ```suggestion
Astro Runtime 5.0.7 includes the following bug fixes:
```

> - Fixed an issue where you could not import plugins as Python packages ([#25296](https://github.com/apache/airflow/pull/25296))

Reviewer 2: ```suggestion
- Fixed an issue where webserver would crash loading plugins ([#25296](https://github.com/apache/airflow/pull/25296))
```
A little more clear what the actual impact was.

- Fixed the cleanup for zombie tasks from DAGs with parsing errors ([#25550](https://github.com/apache/airflow/pull/25550))

> - Fixed an issue where clearing a deferred task instance would not clear its `next_method` field ([#23929](https://github.com/apache/airflow/pull/23929))
> 

Reviewer: ```suggestion
- Fixed an issue where clearing a deferred task instance would not clear its `next_method` field ([#23929](https://github.com/apache/airflow/pull/23929))

For the benefit of Astro users, these change were backported from an Apache Airflow release that is not yet generally available.
```

### Additional improvements

> - Upgraded `openlineage-airflow` to `0.12.0`, which includes support for Spark 3.3.0 and Apache Flink. For a list of all changes, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

Reviewer: ```suggestion
- Upgraded `openlineage-airflow` to version `0.12.0`, which includes support for Spark 3.3.0 and Apache Flink. For a list of all changes, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).
```

> - Upgraded `astronomer-providers` to `1.7.1`, which includes new deferrable operators and improvements to documentation. For more information, see the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.7.1/CHANGELOG.rst).

Reviewer: ```suggestion
- Upgraded `astronomer-providers` to version `1.7.1`, which includes new deferrable operators and improvements to documentation. For more information, see the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.7.1/CHANGELOG.rst).
```

> - Upgraded `apache-airflow-providers-amazon` to `4.1.0`, which includes a bug fix for integrating with AWS Secrets Manager.

Reviewer: ```suggestion
- Upgraded `apache-airflow-providers-amazon` to version `4.1.0`, which includes a bug fix for integrating with AWS Secrets Manager.
```

To upgrade Astro Runtime, see [Upgrade Astro Runtime](upgrade-runtime.md). For general product release notes, see [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, contact [Astronomer support](https://support.astronomer.io).

## Astro Runtime 5.0.7

- Release date: August 16, 2022
- Airflow version: 2.3.3

### Backported fixes from Apache Airflow

Astro Runtime 5.0.7 includes the following bug fixes:

- Fixed an issue where [plugins specified as a python package](https://airflow.apache.org/docs/apache-airflow/stable/plugins.html#plugins-as-python-packages) in an `entry_points` argument were incorrectly loaded twice by Airflow and resulted in an error in the Airflow UI. This bug was introduced in Airflow 2.3.3 ([#25296](https://github.com/apache/airflow/pull/25296))
- Fixed an issue where zombie tasks were not properly cleaned up from DAGs with parse errors [#25550](https://github.com/apache/airflow/pull/25550))
- Fixed an issue where clearing a deferred task instance would not clear its `next_method` field ([#23929](https://github.com/apache/airflow/pull/23929))

> These changes were backported from an Apache Airflow release that is not yet generally available. The bug fixes were also backported to Astro Runtime 5.0.5.

Reviewer: ```suggestion
These changes were backported from Apache Airflow 2.3.4, which is not yet generally available. The bug fixes were also backported to Astro Runtime 5.0.5.
```
